metrix/src/main.rs

```diff
@@ -41,6 +44,73 @@ fn create_metric_route(metric_body: Json<NewMetric>) -> Json<Metric> {
     Json(result)
 }
 
+#[get("/?<offset>&<start_datetime>&<end_datetime>")]
+fn query_metric_route(
+    offset: Option<&RawStr>,
+    start_datetime: Option<&RawStr>,
+    end_datetime: Option<&RawStr>,
+) -> Json<Vec<Metric>> {
+    let db_conn = establish_connection();
+    let mut filter_clause = String::from("WHERE 1=1");
+
+    if offset.is_some() {
+        let result = offset.unwrap().url_decode();
+        // https://api.rocket.rs/v0.3/rocket/http/struct.RawStr.html
+        if result.is_ok() {
+            let metric_id_str: String = result.ok().unwrap();
+            filter_clause.insert_str(
+                filter_clause.len(),
+                &format!(" AND id > {}", metric_id_str).to_string()
+            );
+        }
+    }
+
+    if start_datetime.is_some() {
+        let result = start_datetime.unwrap().url_decode();
+        if result.is_ok() {
+            let created_at_start = result.ok().unwrap();
+            let created_at_start_parsed = NaiveDateTime::parse_from_str(
+                &created_at_start,
+                &"%Y-%m-%dT%H:%M:%S".to_string() // "2014-5-17T12:34:56"
+            );
+
+            if created_at_start_parsed.is_ok() {
+                filter_clause.insert_str(
+                    filter_clause.len(),
+                    &format!(" AND created_at > '{}'", created_at_start).to_string()
+                );
+            }
+        }
+    }
+
+    if end_datetime.is_some() {
+        let result = end_datetime.unwrap().url_decode();
+        if result.is_ok() {
+            let created_at_end = result.ok().unwrap();
+            let created_at_end_parsed = NaiveDateTime::parse_from_str(
+                &created_at_end,
+                &"%Y-%m-%dT%H:%M:%S".to_string() // "2014-5-17T12:34:56"
+            );
+
+            if created_at_end_parsed.is_ok() {
+                filter_clause.insert_str(
+                    filter_clause.len(),
+                    &format!(" AND created_at < '{}'", created_at_end).to_string()
+                );
+            }
+        }
+    }
+
+    println!("### QUERY: SELECT * FROM metrics {};", filter_clause);
+
+    let query_string = format!("SELECT * FROM metrics {} ORDER BY id LIMIT 10", filter_clause);
```

Comment:
Any reason for limit 10?

Reply:
to test pagination, we can change it, nbd.